### DISCUSSION TITLE
Faster RCNN validation step, reasoning for `box_iou`

### DISCUSSION THREAD

**Discussion by vahvero** [PyTorchLightning/pytorch-lightning-bolts#721] [Ideas]
Currently `FasterRCNN`'s `validation_step` is based on `box_iou`. 

```python
    def validation_step(self, batch, batch_idx):
        images, targets = batch
        # fasterrcnn takes only images for eval() mode
        outs = self.model(images)
        iou = torch.stack([_evaluate_iou(t, o) for t, o in zip(targets, outs)]).mean()
        return {"val_iou": iou}

    def validation_epoch_end(self, outs):
        avg_iou = torch.stack([o["val_iou"] for o in outs]).mean()
        logs = {"val_iou": avg_iou}
        return {"avg_val_iou": avg_iou, "log": logs}
```

Shouldn't it be better to use validation loss as suggested in [torch forums](https://discuss.pytorch.org/t/compute-validation-loss-for-faster-rcnn/62333) as most of the user probably want to use early stopping with loss rather than `box_iou`?

I've currently overridden these methods but was wondering why it had been done this way.

My current unrefined code:

```python
    def validation_step(self, batch, batch_idx):
        images, targets = batch
        with torch.no_grad():
            self.model.train()
            outs = self.model(images, targets)
            losses = sum(outs.values())
            loss = losses * len(images)
           # For early stopping callback
            self.log("val_loss", loss)
        return {"val_loss": loss}

    def validation_epoch_end(self, outs):
        avg_loss = torch.stack([o["val_loss"] for o in outs]).mean()
        logs = {"avg_val_loss": avg_loss}
        return {"avg_val_loss": avg_loss, "log": logs}
```
